lighthouse-cli/test/smokehouse/readme.md

```diff
@@ -0,0 +1,127 @@
+
+### Special numeric expectations
+
+If checking a number somewhere in the Lighthouse results, numeric comparisons can be used in place of a raw expected number. This allows asserting ranges or categories of numbers where the exact value isn't necessarily important, or to allow for expected variability in a test.
+
+The comparator is specified with a string, and the actual value being tested must be a number. Whitespace may be included in the string for readability.
+
+The following operators are supported:
+
+| Operator | Example|
+| -- | -- |
+| `>` | `'>0'` |
+| `>=` | `'>=5'` |
+| `<` | `'<1'` |
+| `<=` | `'<=10'` |
+| `+/-` | `'100+/-10'` |
+| `±` | `'100±10'` |
+
+### Special string expectations
+
+If checking a string somewhere in the Lighthouse results, a regular expression can be used in place of a string literal.
+
+### Special array expectations
+
+Individual elements of an array can be asserted by using numeric properties in an object, e.g. asserting the third element in an array is 5: `{2: 5}`.
+
+However, if an array literal is used as the expectation, an extra condition is enforced that the actual array *must* have the same length as the provided expected array.
+
+## Pipeline
+
+The different frontends launch smokehouse with a set of tests to run. Smokehouse then coordinates the tests using a particular method of running Lighthouse (CLI, as a bundle, etc).
+
+```
+Frontends                                                   Lighthouse Runners
++------------+
+|            |
+|  bin (CLI) +----+                                           +--------------+
+|            |    |                                           |              |
++------------+    |                                       +-->+     cli      |
+                  |                                       |   |              |
++------------+    |            +---------------+          |   +--------------+
+|            |    | testDefns> |               |  config> |
+|   node     +---------------->+ smokehouse.js +<---------+
+|            |    |            |               |   <lhr   |   +--------------+
++------------+    |            +-------+-------+          |   |              |
+                  |                    ^                  +-->+   bundled    |
++------------+    |                    |                      |              |
+|            |    |                    |                      +--------------+
+|bundle+entry+----+                    v
+|            |                +--------+--------+
++------------+                |                 |
+                              |  report/assert  |
+                              |                 |
+                              +-----------------+
+```
+
+### Frontends
+- `frontends/smokehouse-bin.js` - runs smokehouse from the command line
+- TODO: bundle-entry - simple entrypoint to smokehouse for bundling and running in a browser.
+- `smokehouse.js` - smokehouse itself is runnable from node
```

Comment:
having this here in this list but not in the folder threw me for a loop. also smokehouse.js listed here and in the next section. 😕 

can you add a `frontend/node.js` that just `module.exports = require('../smokehouse.js');`  ? 

that'd help my mental model of this thing.